frontend/src/metabase/visualizations/lib/renderer_utils.js

```diff
@@ -352,6 +352,72 @@ export function xValueForWaterfallTotal({ settings, series }) {
   return TOTAL_ORDINAL_VALUE;
 }
 
+export function shouldSplitYAxis(
+  { settings, chartType, isScalarSeries, series },
+  datas,
+  yExtents,
+) {
+  if (
+    isScalarSeries ||
+    chartType === "scatter" ||
+    settings["graph.y_axis.auto_split"] === false ||
+    isStacked(settings, datas)
+  ) {
+    return false;
+  }
+
+  const hasDifferentYAxisColTypes =
+    _.uniq(series.map(s => s.data.cols[1].semantic_type)).length > 1;
+  if (hasDifferentYAxisColTypes) {
+    return true;
+  }
+
+  const columnSettings = series.map(s =>
+    normalizeLineAreaBarSettings(
+      settings.column_settings[getColumnSettingsKey(s)],
+    ),
+  );
+  const hasDifferentColumnSettings = columnSettings.some(s1 =>
+    columnSettings.some(s2 => !_.isEqual(s1, s2)),
+  );
+  if (hasDifferentColumnSettings) {
+    return true;
+  }
+
+  const minRange = Math.min(...yExtents.map(extent => extent[1] - extent[0]));
+  const maxExtent = Math.max(...yExtents.map(extent => extent[1]));
+  const minExtent = Math.min(...yExtents.map(extent => extent[0]));
+  const chartRange = maxExtent - minExtent;
+
+  // Note (EmmadUsmani): When the series with the smallest range is less than 5%
+  // of the chart's total range, we split the y-axis so it doesn't look too small.
+  return minRange / chartRange <= 0.05;
+}
+
+function getColumnSettingsKey(series) {
+  return JSON.stringify(["name", series.data.cols[1].name]);
+}
+
+function normalizeLineAreaBarSettings(colSettings) {
+  const defaultSettings = {
```

Comment:
I'd prefer not to duplicate `defaultSettings` here because they can get out of sync. You could get already normalized column settings like this
`const columnSettings = series.map(s => settings.column(s.data.cols[1]));`

Reply:
I agree it's better not to duplicate `defaultSettings` and to use `settings.column` instead. However `settings.column` doesn't return a perfectly normalized version of the column settings which I need for comparison.

<img width="775" alt="Screenshot 2023-03-16 at 8 59 45 AM" src="https://user-images.githubusercontent.com/37751258/225683446-a758d82f-091d-462a-8f70-32332e1e0f33.png">

First, it returns extra fields (`_column_title_full`, `_numberFormatter`, and `column`) that mess up the comparison, so I have to manually remove them. This is not a huge issue.

<img width="667" alt="Screenshot 2023-03-16 at 9 07 33 AM" src="https://user-images.githubusercontent.com/37751258/225683809-9f454108-7e13-44e9-90f6-7432b362364f.png">

The more concerning issue is that it does not remove the strings from `prefix` and `suffix` when they are empty, so I have to manually do that. I don't like this since it also feels like a duplication of the source of truth for column settings, I don't like having to manually delete some specific settings, and there is a potential for bugs to be introduced if new settings are added that are not deleted when they return to their default value.

For now I've updated my function to look like this
```javascript
function normalizeColumnSettings(settings, series) {
  const colSettings = { ...settings.column(series.data.cols[1]) };
  delete colSettings._column_title_full;
  delete colSettings._numberFormatter;
  delete colSettings.column;
  delete colSettings.prefix;
  delete colSettings.suffix;

  return colSettings;
}
```
but let me know if you think there is a better way to do this. Perhaps we should update the `.column` method to not return `prefix` and `suffix` when they are empty strings?

Reply:
Yes, we need to compare only specific keys. What do you think about:
1) creating a util function in `frontend/src/metabase/visualizations/lib/settings.js` like this:
```
const COLUMN_SETTINGS_KEYS_TO_COMPARE = [
  "prefix",
  "suffix",
  "currency",
  "number_style",
  "number_separators",
  // ...
];

function getColumnFormattingSettings(columnSettings) {
  return _.pick(columnSettings, (value, key) => {
    if (!COLUMN_SETTINGS_KEYS_TO_COMPARE.includes(key)) {
      return false;
    }

    if (key === "suffix" && value === "") {
      return false;
    }

    // handle other settings

    return true;
  });
}
```

2) Using it here
```
const columnSettings = series.map(s =>
  getColumnFormattingSettings(settings.column(s.data.cols[1])),
);

```

We cannot really delete `prefix` and `suffix` because they can be different for each column.

The previous way with `defaultSettings` would not work correctly because actual defaults are selected in the admin panel:

<img width="399" alt="Screen Shot 2023-03-16 at 2 17 36 PM" src="https://user-images.githubusercontent.com/14301985/225700162-539489f7-0ed4-4a17-8f2a-8d2f5a23b522.png">

Reply:
This looks great, thanks! I like that the code is close to the source of truth in `visualizations/lib/settings.js`. I will update the PR with something like this.